### PR TITLE
fix: output better serde::yaml errors

### PR DESCRIPTION
**Pull request by bayou-brogrammer**

### CODE REVIEW

**Comment by bayou-brogrammer** [comment]
image

**Comment by gmpinder** [comment]
Wasn't expecting this so quick!

File: src/commands/build.rs [hunk 1]
```diff
@@ -196,7 +196,8 @@ impl BuildCommand {
             bail!("Failed to get credentials");
         }
 
-        let recipe: Recipe = serde_yaml::from_str(fs::read_to_string(recipe_path)?.as_str())?;
+        let recipe: Recipe = serde_yaml::from_str(fs::read_to_string(recipe_path)?.as_str())
+            .map_err(|err| SerdeError::new(recipe_path.display().to_string(), err))?;
```
Comment (gmpinder) [inline src/commands/build.rs:200]: I'm pretty sure you need to pass in the file contents here.

File: src/commands/build.rs [hunk 2]
```diff
@@ -280,7 +281,8 @@ impl BuildCommand {
     fn build_image(&self, recipe_path: &Path) -> Result<()> {
         trace!("BuildCommand::build_image()");
 
-        let recipe: Recipe = serde_yaml::from_str(fs::read_to_string(recipe_path)?.as_str())?;
+        let recipe: Recipe = serde_yaml::from_str(fs::read_to_string(recipe_path)?.as_str())
+            .map_err(|err| SerdeError::new(recipe_path.display().to_string(), err))?;
```
Comment (gmpinder) [inline src/commands/build.rs:285]: Same here

File: Cargo.toml [hunk 3]
```diff
@@ -20,6 +20,7 @@ colorized = "1"
 derive_builder = "0.13"
 directories = "5"
 env_logger = "0.11"
+format_serde_error = { git = "https://github.com/duyet/format_serde_error" }
```
Comment (gmpinder) [inline Cargo.toml:23]: I think we should be using a published crate instead, unless there's a change you're wanting to use that isn't published yet

**Review by gmpinder — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 3 above)
Comment (bayou-brogrammer) [reply Cargo.toml:23]: The core crate is unmaintained and abandoned. So we either find a new crate or fork it and maintain it ourselves

Comment (gmpinder) [reply Cargo.toml:23]: Oh really? Well that sucks. What changes in this git version are there that we're using and doesn't exist in the published one?

Comment (bayou-brogrammer) [reply Cargo.toml:23]: They just bump serde yaml to 0.9

Comment (gmpinder) [reply Cargo.toml:23]: Ah ok, yeah that's fairly important lol

**Comment by bayou-brogrammer** [comment]
@gmpinder i forked serde_format_error and fixed it up on our side. We can either maintain it, or find a new crate!

File: src/commands/local.rs [hunk 4]
```diff
@@ -44,7 +45,9 @@ impl BlueBuildCommand for UpgradeCommand {
         check_can_run()?;
 
         let recipe: Recipe =
-            serde_yaml::from_str(fs::read_to_string(&self.common.recipe)?.as_str())?;
+            serde_yaml::from_str(fs::read_to_string(&self.common.recipe)?.as_str())
+                .map_err(|err| SerdeError::new(self.common.recipe.display().to_string(), err))?;
```
Comment (gmpinder) [inline src/commands/local.rs:49]: Need file contents here too

File: src/commands/local.rs [hunk 5]
```diff
@@ -91,7 +94,9 @@ impl BlueBuildCommand for RebaseCommand {
...
```
Comment (gmpinder) [inline src/commands/local.rs:98]: And here

**Review by gmpinder — CHANGES_REQUESTED** [review]

**Comment by gmpinder** [comment]
> @gmpinder i forked serde_format_error and fixed it up on our side. We can either maintain it, or find a new crate!

Shouldn't be too hard to maintain. Would probably only need to keep versions up to date

**Review by gmpinder — APPROVED** [review]

**Pull request merged**